### ISSUE TITLE
bun repl crash

### ISSUE THREAD

**Issue by xicabin** [oven-sh/bun#12086]
### How can we reproduce the crash?

Run `bun repl`

### Relevant log output

```shell
darcy → bun repl 
============================================================
Bun v1.1.16 (bf7b327f) macOS Silicon
Args: "bun" "bun-repl"
Elapsed: 1ms | User: 4ms | Sys: 2ms
RSS: 2.92MB | Peak: 2.92MB | Commit: 1.07GB | Faults: 9

panic(main thread): Bus error at address 0x104ADDBE0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.16/MB1bf7b327AA__+h+lEunlB2pY__A4Cg+t31E

zsh: trace trap  bun repl
```

### Stack Trace (bun.report)

Bun v1.1.16 ([`bf7b327`](<https://github.com/oven-sh/bun/tree/bf7b327f68568104a5e40c1944b7b255b02f0745>)) on macos aarch64 [BunxCommand]

Bus error at address 0x10389DBE0

- *2 unknown/js code*
- `src.cli.bunx_command.BunxCommand.getBinNameFromTempDirectory`
- `src.cli.Cli.start`
- `_mh_execute_header`
- *2 unknown/js code*

**Comment by Jarred-Sumner** [comment]
Duplicate of #12070

**Comment by Jarred-Sumner** [comment]
Fixed in #12071 

to upgrade, run `bun upgrade --canary`
